jans-auth-server/common/src/test/java/io/jans/as/common/RedirectUriTest.java

```diff
@@ -0,0 +1,342 @@
+
+@Listeners(MockitoTestNGListener.class)
+public class RedirectUriTest {
+
+    @Mock
+    private AbstractCryptoProvider cryptoProvider;
+
+    @Test
+    public void parseQueryString_queryStringNull_EmptyResult() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.parseQueryString(null);
+        assertEquals(redirectUri.getResponseParamentersSize(), 0);
+    }
+
+    @Test
+    public void parseQueryString_oneParam_size1() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.parseQueryString(AuthorizeResponseParam.EXPIRES_IN + "=");
+        assertEquals(redirectUri.getResponseParamentersSize(), 1);
+    }
+
+    @Test
+    public void parseQueryString_threeParams_size3() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.parseQueryString(AuthorizeResponseParam.CODE + "=Pcode&" + AuthorizeResponseParam.EXPIRES_IN + "=3000&" + AuthorizeResponseParam.ACCESS_TOKEN + "=tk123");
+        assertEquals(redirectUri.getResponseParamentersSize(), 3);
+    }
+
+    @Test
+    public void parseQueryString_threeParamsOneNull_size3() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.parseQueryString(AuthorizeResponseParam.CODE + "=Pcode&" + AuthorizeResponseParam.EXPIRES_IN + "=&" + AuthorizeResponseParam.ACCESS_TOKEN + "=tk123");
+        assertEquals(redirectUri.getResponseParamentersSize(), 3);
+    }
+
+    @Test
+    public void getQueryString_noResponseMode_responseParametersEncoded() throws UnsupportedEncodingException, CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(null, null, null, null);
+        redirectUri.addResponseParameter(AuthorizeResponseParam.CODE, "123");
+        redirectUri.addResponseParameter(AuthorizeResponseParam.EXPIRES_IN, "3000");
+        redirectUri.addResponseParameter(AuthorizeResponseParam.ACCESS_TOKEN, "tk123");
+        String queryResult = redirectUri.getQueryString();
+
+        assertNoEmptyQueryString(queryResult, AuthorizeResponseParam.CODE, 3);
+    }
+
+    @Test()
+    public void getQueryString_responseModeFormPostJWT_NoEmpty() throws CryptoProviderException {
+        RedirectUri redirectUri = getRedirectUriTemplateGetQueryString(ResponseMode.FORM_POST_JWT, null, null, null);
+        AbstractCryptoProvider cryptoProvider = mock(AbstractCryptoProvider.class);
+        when(cryptoProvider.sign(anyString(), anyString(), anyString(), Mockito.any())).thenReturn("12345");
+        redirectUri.setCryptoProvider(cryptoProvider);
+        String queryResult = redirectUri.getQueryString();
+        //No empty Result
+        assertTrue(queryResult.length() > 0);
```

Comment:
I'd suggest to add `verify` sentences at the end for CryptoProvider as well. Verify that certain methods were called with specific values.